lib/encoder-stream.js

```diff
@@ -2,11 +2,13 @@ const {Transform} = require('stream');
 const beamcoder = require('beamcoder');
 
 class EncoderStream extends Transform {
-	constructor(encoderParameters) {
+	constructor({encoderParameters, fps = 25}) {
```

Comment:
I would prefer no default value is defined here and we raise an error if fps is not defined (and keyframeOnly is false).

Reply:
@piercus 
Why do we need the Producer timestamp? As I see in the frame object we get the pts timestamp which is already unique for every frame. Can't we save the file with pts as filename?

Reply:
AWS_KINESISVIDEO_PRODUCER_TIMESTAMP is a more reliable information.
pts seems a good information to compare the time between 2 frames, but we do not know how reliable is it and which time does it represent : 
* It it a producer timestamp ?
* Is it the receiver timestamp ?
* Is it the timestamp from the beginning of the stream ?

The producerTimestamp has a clear definition, it is the time, since epoch, when the producer produced this exact block of data.